src/index.js

```diff
@@ -250,64 +250,76 @@ export default class CanvasDraw extends PureComponent {
-   * @param {string} fileType Specifies the file format to export to. Note: should only be the file type, not the "image/" prefix.
-   * @param {string} output Specifies whether the function creates an image out of the background image or the drawing. 
-   * It puts the drawing on top of the image when output is undefined
+    canvas.width = width;
+    canvas.height = height;
 
-   */
+    return canvas.getContext("2d");
+  };
 
-  getDataUrl = (fileType, output) => {
-    if (!fileType) fileType = "png";
+  drawInpainting = (imageExportContext, maskExportContext) => {
+    const exportDrawingSize = this.getDrawingExportSize();
 
-    const oldView = this.coordSystem.view;
-    this.resetView();
+    this.drawExportItem(
+      imageExportContext,
+      exportDrawingSize,
+      this.canvas.grid
+    );
+    this.drawExportItem(
+      maskExportContext,
+      exportDrawingSize,
+      this.canvas.drawing
+    );
+    this.convertToMask(maskExportContext, exportDrawingSize);
+  };
 
-    const exportCanvasSize = this.getExportCanvasSize();
+  drawOutpainting = (imageExportContext, maskExportContext) => {
     const exportDrawingSize = this.getDrawingExportSize();
 
-    const exportCanvas = document.createElement("canvas");
-    exportCanvas.width = exportCanvasSize.width;
-    exportCanvas.height = exportCanvasSize.height;
-    const exportContext = exportCanvas.getContext("2d");
+    this.drawExportItem(
+      imageExportContext,
+      exportDrawingSize,
+      this.canvas.drawing
+    );
+    this.drawExportItem(
+      imageExportContext,
+      exportDrawingSize,
+      this.canvas.grid
+    );
 
-    if (!output || output === "image" || output === "imageWithDrawing") {
-      this.drawExportItem(exportContext, exportDrawingSize, this.canvas.grid);
-    }
+    this.drawExportItem(maskExportContext, exportDrawingSize, this.canvas.grid);
+    this.convertToMask(maskExportContext, exportDrawingSize, {
+      transparent: 255,
+      default: 0,
+    });
+  };
 
-    if (!output || output === "mask" || output === "imageWithDrawing") {
-      this.drawExportItem(
-        exportContext,
-        exportDrawingSize,
-        this.canvas.drawing
-      );
+  getImages = (output) => {
+    const oldView = this.coordSystem.view;
+    this.resetView();
 
-      if (output === "mask") {
-        this.convertToMask(exportContext, exportDrawingSize);
-      }
-    }
+    const imageExportCanvas = document.createElement("canvas");
+    const maskExportCanvas = document.createElement("canvas");
+    const imageExportContext =
+      this.createExportCanvasContext(imageExportCanvas);
+    const maskExportContext = this.createExportCanvasContext(maskExportCanvas);
 
-    if (!output || output === "outpaint-mask") {
-      // Draw image with the artboard
-      this.drawExportItem(exportContext, exportDrawingSize, this.canvas.grid);
+    if (output === "inpaint") {
```

Comment:
I think would be better to have an enum for the different options, which could be exported for external use when passing the output prop ("inpaint" or "outpaint")